server/logging/log.go

```diff
@@ -133,8 +133,14 @@ func (l *log) Snapshot(c context.Context, path string, w io.Writer) error {
 	}
 	s.Lock()
 	for _, entry := range s.list {
```

Comment:
I think we can use `defer` instead of adding it in front of every return.
```suggestion
        defer s.unlock()
	for _, entry := range s.list {
```